app/src/main/java/meme5/c4q/nyc/meme_project/MainActivity.java

```diff
@@ -1,15 +1,65 @@
 package meme5.c4q.nyc.meme_project;
 
 import android.app.Activity;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
 import android.os.Bundle;
-
+import android.view.View;
 
 public class MainActivity extends Activity {
 
+    private static final int REQUEST_IMAGE_CAPTURE = 1;
+    private String imgFilePath;
+    private boolean imageSelected;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        imageSelected = false;
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if(imageSelected){
+            launchChooseMeme();
+        }
+    }
+
+    private void launchChooseMeme(){
+        Intent chooseMeme = new Intent(this,ChooseMemeStyle.class);
+        chooseMeme.putExtra("imgFilePath",imgFilePath);
+        startActivity(chooseMeme);
+    }
+
+    public void takePicture(View view){
+
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+            imageSelected = true;
```

Comment:
instead of creating an imageSelected variable, do you think it's ok to just call `launchChooseMeme()` here (and then also get rid of lines 24 - 30) or does it have to be done on onResume()?